pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -351,17 +408,36 @@ func TestProcessResults(t *testing.T) {
 				defaultFS = afero.NewOsFs()
 			}()
 
-			err := processResults(tt.resultMap, option.dir)
-			if tt.expectedErr != "" {
-				require.ErrorContains(t, err, tt.expectedErr)
-			} else {
-				require.NoError(t, err)
-			}
-			// Both test cases above have failed Nodes, hence this file should always be created/
+			antreaInterface := fakeclientset.NewSimpleClientset(&controllerInfo, agentInfo1, agentInfo2)
+			k8sClient := fake.NewSimpleClientset(controllerPod, pod1, pod2)
+			err := processResults(context.TODO(), antreaInterface, k8sClient, tt.resultMap, option.dir)
+			require.NoError(t, err)
 			b, err := afero.ReadFile(defaultFS, filepath.Join(option.dir, "failed_nodes"))
 			require.NoError(t, err)
 			data := string(b)
 			for node, err := range tt.resultMap {
+				tgzFileName := fmt.Sprintf("agent_%s.tar.gz", node)
+				if node == "" {
+					tgzFileName = "controller_node-1.tar.gz"
+				}
+				if err != nil {
+					ok, checkErr := afero.Exists(defaultFS, filepath.Join(option.dir, tgzFileName))
+					require.NoError(t, checkErr)
+					require.True(t, ok, "expected support bundle file %s not found", tgzFileName)
+
+					unpackError := compress.UnpackDir(defaultFS, filepath.Join(option.dir, tgzFileName), option.dir)
+					require.NoError(t, unpackError)
+					expectFileName := "logs/agent/antrea-agent.log"
+					if node == "" {
+						expectFileName = "logs/controller/antrea-controller.log"
+					}
+					ok, checkErr = afero.Exists(defaultFS, filepath.Join(option.dir, expectFileName))
+					require.NoError(t, checkErr)
+					assert.True(t, ok, "expected log file %s not found", expectFileName)
+					deleteErr := defaultFS.Remove(filepath.Join(option.dir, expectFileName))
+					require.NoError(t, deleteErr)
```

Comment:
ditto, this could be a single line

Also, I am not sure it is actually necessary?